argus/config_generator/windows/cb_init.py

```diff
@@ -93,7 +94,11 @@ def _get_service(self, service_type):
             This can be HTTP, ConfigDrive, EC2, OpenNebula,
             CloudStack or MAAS.
         """
-        return '.'.join([util.SERVICES_PREFIX, self.SERVICES[service_type]])
+        if self.SERVICES[service_type]:
+            return
```

Comment:
Because you return `None` the join at line 121 might fail, return an empty string.